modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EDecimal.java

```diff
@@ -55,48 +50,4 @@ public String getDecimal() {
     public <Scope> void visitChildren(UserTreeVisitor<Scope> userTreeVisitor, Scope scope) {
         // terminal node; no children
     }
-
-    @Override
-    void analyze(SemanticScope semanticScope) {
-        analyze(semanticScope, false);
-    }
-
-    void analyze(SemanticScope semanticScope, boolean negate) {
-        if (semanticScope.getCondition(this, Write.class)) {
```

Comment:
I like the new use of `Negate` condition in DefaultSemanticAnalysisPhase.

Reply:
:)